dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+        return active_channels
+
+
+    def get_number_of_streams(self, channel):
+        stream_count = 0
+        cf = self.freq_dict[channel]
+        for i in range(3):
+            try:
+                request = '.node-config.ch'+str(i)+'.strw.center-freq'
+                result = self.provider.set(self.queue_dict[channel]+request, cf)
+                stream_count += 1
+                self.mode_dict[channel]='streaming'
+            except:
+                try:
+                    request = '.node-config.ch'+str(i)+'.trw.center-freq'
+                    result = self.provider.set(self.queue_dict[channel]+request, cf)
+                    stream_count += 1
+                    self.mode_dict[channel]='triggered'
+                except:
+                    pass  
+        logger.info('Number of streams for channel {}: {}'.format(channel, stream_count))
+        return stream_count
+
+
+    def set_fmt_snr_threshold(self, channel='a', snr=6):
+        if self.mode_dict[channel] == 'streaming':
+            logger.warning('Psyllid not in streaming mode')
+            return False
+#        request = 'active-config.ch'+str(self.channel_dict[channel])+'.fmt.threshold-power'
+        request = '.node-config.ch'+str(self.channel_dict[channel])+'.fmt.threshold-power-snr'
+        result = self.provider.set(self.queue_dict[channel]+request, snr)
+        self.deactivate(channel)
+        time.sleep(1)
+        self.activate(channel)
+        return True
+
+
+    def make_trigger_mask(self, channel='a', filename='~/fmt_mask.json'):
+        if self.mode_dict[channel] == 'streaming':
+            return False
+
+        self.request_status(channel)
+        if self.status_value_dict[channel] !=4:
+            self.activate(channel)
+
+        logger.info('freq-only')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.freq-only'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+
+        logger.info('update-mask')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.update-mask'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+        time.sleep(1)
+
+        logger.info('run')
+        request = 'start-run'
+        result = self.provider.cmd(self.queue_dict[channel], request)
+        time.sleep(1)
+
+        logger.info('write-mask')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.write-mask'
+        payload = {'filename': filename}
+        result = self.provider.cmd(self.queue_dict[channel], request, payload=payload)
+
+        logger.info('back to time and freq')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.time-and-freq'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+
+        logger.info('apply-trigger')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.apply-trigger'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+  
```

Comment:
`provider.[get,set,cmd]` uses `provider._send_request`, which has a built in check for error codes + raise exception; as long as these aren't embedded in try-except loops won't that be sufficient?